File: CMS/src/app/shared/PhanAnh.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { PhanAnh } from 'src/app/shared/PhanAnh.model';
import { BaseService } from './Base.service';
@Injectable({
  providedIn: 'root'
})
export class PhanAnhService extends BaseService {


  List: PhanAnh[] | undefined;
  ListFilter: PhanAnh[] | undefined;
  FormData!: PhanAnh;

  DisplayColumns001: string[] = ['STT', 'NgayGhiNhan', 'Name', 'Display', 'fullname', 'phone', 'email'];
  DisplayColumns002: string[] = ['STT', 'NgayGhiNhan', 'Name', 'HTMLContent', 'Description', 'Save'];
  DisplayColumns003: string[] = ['STT', 'NgayGhiNhan', 'Display', 'Name', 'HTMLContent', 'Description'];

  constructor(public httpClient: HttpClient) {
    super(httpClient);
    this.Controller = "PhanAnh";
  }

  GetByParentIDToListAsync() {
    let url = this.APIURL + this.Controller + '/GetByParentIDToListAsync';
    if (this.BaseParameter.ParentID == 0) {
      return this.GetAllToListAsync();
    }
    const formUpload: FormData = new FormData();
    formUpload.append('data', JSON.stringify(this.BaseParameter));
    return this.httpClient.post(url, formUpload, { headers: this.Headers });
  }
}

File: Service/Implement/PhanAnhService.cs
using Service.Interface;

namespace Service.Implement
{
	public class PhanAnhService : BaseService<PhanAnh, IPhanAnhRepository>
	, IPhanAnhService
	{
		private readonly IPhanAnhRepository _PhanAnhRepository;

		private readonly ICompanyInfoService _CompanyInfoService;
		public PhanAnhService(IPhanAnhRepository PhanAnhRepository

			, ICompanyInfoService CompanyInfoService

			) : base(PhanAnhRepository)
		{
			_PhanAnhRepository = PhanAnhRepository;

			_CompanyInfoService = CompanyInfoService;
		}
		public override void Initialization(PhanAnh model)
		{
            BaseInitialization(model);
            if (model.ParentID > 0)
			{
				model.Display = _CompanyInfoService.GetByID(model.ParentID.Value).Name;
			}
			if (model.NgayGhiNhan == null)
			{
				model.NgayGhiNhan = GlobalHelper.InitializationDateTime;
			}
		}
	}
}

